features/fixtures/mazerunner/jvm-scenarios/src/main/java/com/bugsnag/android/mazerunner/scenarios/FeatureFlagScenario.kt

```diff
@@ -0,0 +1,31 @@
+package com.bugsnag.android.mazerunner.scenarios
+
+import android.content.Context
+import com.bugsnag.android.Bugsnag
+import com.bugsnag.android.Configuration
+
+class FeatureFlagScenario(
+    config: Configuration,
+    context: Context,
+    eventMetadata: String
+) : Scenario(config, context, eventMetadata) {
+    override fun startScenario() {
+        super.startScenario()
+
+        Bugsnag.addFeatureFlag("demo_mode")
+        Bugsnag.addFeatureFlag("should_not_be_reported")
+
+        if (eventMetadata?.contains("callback") == true) {
+            Bugsnag.addOnError { error ->
+                error.addFeatureFlag("sample_group", "a")
+                return@addOnError true
+            }
+        }
+
+        if (eventMetadata?.contains("unhandled") == true) {
+            throw RuntimeException("FeatureFlagScenario unhandled")
```

Comment:
It'd be good to add a test to ensure that the functionality works for unhandled NDK errors too.

Reply:
The native functionality was added in its own scenario (batch_1 / batch_2). I'll rebase and merge them together.